Test/comp/UnicodeStrings.dfy

```diff
@@ -0,0 +1,166 @@
+// RUN: %testDafnyForEachCompiler "%s" -- /unicodeChar:1
+
+newtype uint8 = x: int | 0 <= x < 0x100 
+newtype uint16 = x: int | 0 <= x < 0x1_0000 
+newtype uint32 = x: int | 0 <= x < 0x1_0000_0000 
+newtype uint64 = x: int | 0 <= x < 0x1_0000_0000_0000_0000 
+newtype int8 = x: int | -0x80 <= x < 0x80
+newtype int16 = x: int | -0x8000 <= x < 0x8000
+newtype int32 = x: int | -0x8000_0000 <= x < 0x8000_0000
+newtype int64 = x: int | -0x8000_0000_0000_0000 <= x < 8000_0000_0000_0000
+
+// TODO: Test cases for escapes in character literals
+// TODO: Remove token parameter in compilers if not needed
```

Comment:
Somewhere, also include tests that
* check that `\U10000` is parsed correctly (it may not be with the code above)
* check that `\U{E000} - 'a'` gives an error (it may not be with the code above)
* check that `\U{D000} + \U{1000}` gives an error

Reply:
* `\U1_0000` is now a boundary case (along with several others) in UnicodeStrings.dfy
* `Test/unicodechars/dafny0/Char.dfy` now includes the missing verification cases for hitting the surrogate range